stdlib/influxdata/influxdb/monitor/monitor.flux

```diff
@@ -20,14 +20,29 @@ from = (start, stop=now(), fn=(r) => true) =>
         |> filter(fn: fn)
         |> v1.fieldsAsCols()
 
-// StateChanges takes a stream of tables, fromLevel, and toLevel and returns
-// a stream of tables where status has gone from fromLevel to toLevel.
-//
-// StateChanges only operates on data with data containing r._measurement == "statuses"
-// and where r._level exists.
-stateChanges = (fromLevel="any", toLevel, tables=<-) => {
+// levels describing the result of a check
+levelOK = "ok"
+levelInfo = "info"
+levelWarn = "warn"
+levelCrit = "crit"
+levelUnknown = "unknown"
+
+_stateChanges = (fromLevel="any", toLevel="any", tables=<-) => {
+    otherLevelFilter = if toLevel == "any" or fromLevel == "any" then (r) => false
+                   else (r) => r._level != toLevel and r._level != fromLevel
+    
+    otherStatuses = tables
+        |> filter(fn: otherLevelFilter)
+        |> map(fn: (r) => ({r with level_value: -10}))
+        |> duplicate(column: "_level", as: "l2")
```

Comment:
I see it is in the old code as well, but why are these `duplicate-drop-rename` sequences here when they appear to be an identity transformation.

Reply:
Ah I see now, it enforces that `_level` exists, throwing an error if not present. Might be nice to ecapsulate these three calls in a function of their own, so as to reduce the code size and also make their purpose clear. 

One other thought ... if column `l2` exists already there is a problem? Maybe not with the schema this function is intended to work with, but if people try to use it elsewhere?

Reply:
> I see it is in the old code as well, but why are these duplicate-drop-rename sequences here when they appear to be an identity transformation.

Yeah, the big thing we're trying to do here is that we want to get `_level` out of the group key. This was the easiest way we could figure out to do this.

> One other thought ... if column l2 exists already there is a problem? Maybe not with the schema this function is intended to work with, but if people try to use it elsewhere?

That's a really good point. I wonder what a better name would be. Maybe something more obscure like `_____level_temp____`.